homeassistant/components/manual/alarm_control_panel.py

```diff
@@ -203,27 +208,29 @@ class ManualAlarm(AlarmControlPanelEntity, RestoreEntity):
 
     def __init__(
         self,
-        hass,
-        name,
-        code,
-        code_template,
-        code_arm_required,
-        disarm_after_trigger,
-        config,
-    ):
+        hass: HomeAssistant,
+        name: str,
+        unique_id: str | None,
+        code: str | None,
+        code_template: Template | None,
+        code_arm_required: bool,
+        disarm_after_trigger: bool,
+        config: dict[str, Any],
+    ) -> None:
         """Init the manual alarm panel."""
         self._state = STATE_ALARM_DISARMED
         self._hass = hass
         self._attr_name = name
+        self._attr_unique_id = unique_id
         if code_template:
-            self._code = code_template
+            self._code: str | Template | None = code_template
             self._code.hass = hass
         else:
             self._code = code or None
         self._attr_code_arm_required = code_arm_required
         self._disarm_after_trigger = disarm_after_trigger
         self._previous_state = self._state
-        self._state_ts = None
+        self._state_ts: datetime.datetime = dt_util.utcnow()
```

Comment:
Why is this changed?

Reply:
It was mainly to handle typing and is only used once at the very first time as from that point it will use the `last_updated` attribute from the restored state. 
It has no impact and seemed more logical than asserting type-checking or ignoring an assignment.

Reply:
Yes, looks good.